### ISSUE TITLE
Fields assigned with a null-coalescing assignment shouldn't trigger CS8618

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#41307]
**Version Used**: 
```
commit 1d4cf98ece59ab4365adcc7d414d3c758010d698
Merge: 1302d42173 26187bd0d8
Author: Jason Malinowski <jason.malinowski@microsoft.com>
Date:   Mon Jan 13 12:18:17 2020 -0800

    Merge pull request #40897 from jasonmalinowski/null-annotate-quick-info

    Null annotate Quick Info
```

**Steps to Reproduce**:

Compile the following code
```cs
#nullable enable
class C
{
    string s;
    public C()
    {
        s ??= "";
    }
}
```

**Expected Behavior**:
No warnings. `s` is definitely not null by the time the constructor finishes.

**Actual Behavior**:
```
warning CS8618: Non-nullable field 's' is uninitialized. Consider declaring the field as nullable.
```

**Notes**
It seems that Roslyn doesn't consider null-coalescing assignments as assignments when inspeting which fields might end up uninitialized.
I see a couple of problems with this:
- Since it doesn't assign the variable only if it already is initialized there's nothing to warn about.
- It isn't consistent with other possible assignment sources e.g. passing the field as `ref` argument considered as initialization by Roslyn.

The original example is minified to reproduce the problem so you might want to consider a more real exapmles e.g. conditional initialization with a fallback null-coalescing assignment:
```cs
class C
{
   string f;
   public C(int? x)
   {
     if (x != null)
     {
        ProcessX(x.Value);
        if (shouldInitializeFromX(x.Value))
        {
          f = GetStringFromX(x.Value);
        }
     }

     f ??= string.Empty;
   }
}
```

or calling another method which might but not guaranteed to initialize the field:

```cs
class C
{
  string f;
  public C()
  {
    DoSomethingThatMightInitializeFieldF();
    f ??= "";
  }
}
```

Both cases above report warnings for uninitialized non-nullable fields despite being absolutely correct and guaranteed to initialize the field with a non-nullable value.

**Comment by sharwell** [comment]
Duplicate of #41273
